ipatests/test_integration/test_otp.py

```diff
@@ -309,3 +310,58 @@ def test_2fa_disable_single_prompt(self):
             master.run_command(['ipa', 'user-del', USER2])
             self.master.run_command(['semanage', 'login', '-D'])
             sssd_conf_backup.restore()
+
+    @pytest.fixture
+    def setup_otp_nsslapd(self):
+        # setting nsslapd-idletimeout
+        new_limit = 60
+        conn = self.master.ldap_connect()
+        dn = DN(('cn', 'config'))
+        entry = conn.get_entry(dn)  # pylint: disable=no-member
+        orig_limit = entry.single_value.get('nsslapd-idletimeout')
+        ldap_query = textwrap.dedent("""
+            dn: cn=config
+            changetype: modify
+            replace: nsslapd-idletimeout
+            nsslapd-idletimeout: {limit}
+        """)
+        tasks.ldapmodify_dm(self.master, ldap_query.format(limit=new_limit))
+        # Be sure no services are running and failed units
+        self.master.run_command(['killall', 'ipa-otpd'], raiseonerr=False)
+        check_services = self.master.run_command(
+            ['systemctl', 'list-units', '--state=failed']
+        )
+        assert "0 loaded units listed" in check_services.stdout_text
+        assert "ipa-otpd" not in check_services.stdout_text
+        yield
+        # cleanup
+        tasks.ldapmodify_dm(self.master, ldap_query.format(limit=orig_limit))
+
+    def test_check_otpd_after_dirsrv_restart(self, setup_otp_nsslapd):
```

Comment:
Can you change this from restart to something like test_check_otpd_after_idle_timeout ?